web/packages/shared/components/UnifiedResources/UnifiedResources.tsx

```diff
@@ -122,6 +120,23 @@ interface UnifiedResourcesProps {
   availableKinds: SharedUnifiedResource['resource']['kind'][];
   setParams(params: UnifiedResourcesQueryParams): void;
   onLabelClick(label: ResourceLabel): void;
+  /** A list of actions that can be performed on the selected items. */
+  // TODO (avatus): pinning should be removed from the internals of UnifiedResources as passed in
+  // from the parents.
```

Comment:
The longer I think about it, the more convinced I'm that we don't have to remove resource pinning from the `UnifiedResources` component. The main problem is that the cards have a pin button, so we have to pass the information if a resource is pinned and a function to pin/unpin it. In theory, we could do it similarly to `ActionButton`, but it seems unnecessarily complicated.
Because of that, I think that we can remove this TODO.